### ISSUE TITLE
Model estimates criterion

### ISSUE THREAD

**Issue by dbgns** [weiaicunzai/pytorch-cifar100#5]
I have noticed that you use the full training set to train the model and select the best model directly on the test set? This practice will not underestimate the test error ? As I often split a validation set and choose the model based on the performance on the validation set. Thanks in advance

**Comment by weiaicunzai** [comment]
What do you mean by "underestimate the test error"? I will just assume that you are asking me why I choose the best performance model from the test set result, the reason why I did this is just from the engineering perspective, we all want a highest score model .
